pkg/tsdb/cloudwatch/live.go

```diff
@@ -0,0 +1,246 @@
+		return fmt.Errorf("channel with name '%s' already exists", name)
+	}
+
+	responseChannels[name] = channel
+	return nil
+}
+
+func getResponseChannel(name string) (chan *tsdb.Response, error) {
+	channelMu.Lock()
+	defer channelMu.Unlock()
+
+	if responseChannel, ok := responseChannels[name]; ok {
+		return responseChannel, nil
+	}
+
+	return nil, fmt.Errorf("channel with name '%s' not found", name)
+}
+
+// GetHandlerForPath gets called on init.
+func (supplier *LogQueryRunnerSupplier) GetHandlerForPath(path string, publisher models.ChannelPublisher) (models.ChannelHandler, error) {
+	return &LogQueryRunner{
+		channelName: path,
+		publish:     publisher,
+		running:     make(map[string]bool),
+	}, nil
+}
+
+// GetChannelOptions gets channel options.
+// It's called fast and often.
+func (g *LogQueryRunner) GetChannelOptions(id string) centrifuge.ChannelOptions {
+	return centrifuge.ChannelOptions{}
+}
+
+// OnSubscribe for now allows anyone to subscribe to any dashboard
+func (g *LogQueryRunner) OnSubscribe(c *centrifuge.Client, e centrifuge.SubscribeEvent) error {
+	if _, ok := g.running[e.Channel]; !ok {
+		g.running[e.Channel] = true
+		go g.publishResults(e.Channel)
+	}
+
+	return nil
+}
+
+// OnPublish is called when an event is received from the websocket.
+func (g *LogQueryRunner) OnPublish(c *centrifuge.Client, e centrifuge.PublishEvent) ([]byte, error) {
+	return nil, fmt.Errorf("can not publish")
+}
+
+func (g *LogQueryRunner) publishResults(channelName string) error {
+	defer func() {
+		delete(responseChannels, channelName)
+		delete(g.running, channelName)
+	}()
+
+	responseChannel, err := getResponseChannel(channelName)
+	if err != nil {
+		return err
+	}
+
+	for response := range responseChannel {
+		responseBytes, err := json.Marshal(response)
+		if err != nil {
+			return err
+		}
+
+		g.publish(channelName, responseBytes)
+	}
+
+	return nil
+}
+
+func (e *cloudWatchExecutor) executeLiveLogQuery(ctx context.Context, queryContext *tsdb.TsdbQuery) (*tsdb.Response, error) {
+	responseChannelName := uuid.Must(uuid.NewV4()).String()
+	responseChannel := make(chan *tsdb.Response)
+	addResponseChannel("plugin/cloudwatch/"+responseChannelName, responseChannel)
+	requestContext, _ := context.WithTimeout(context.Background(), 15*time.Minute)
+	go e.sendQueriesToChannel(requestContext, queryContext, responseChannel)
+
+	response := &tsdb.Response{
+		Results: map[string]*tsdb.QueryResult{
+			"A": {
+				RefId: "A",
+				Meta: simplejson.NewFromAny(map[string]interface{}{
+					"channelName": responseChannelName,
+				}),
+			},
+		},
+	}
+
+	return response, nil
+}
+
+func (e *cloudWatchExecutor) sendQueriesToChannel(ctx context.Context, queryContext *tsdb.TsdbQuery, responseChannel chan *tsdb.Response) {
+	eg, ectx := errgroup.WithContext(ctx)
+
+	for _, query := range queryContext.Queries {
+		query := query
+		eg.Go(func() error {
+			return e.startQuery(ectx, responseChannel, query, queryContext.TimeRange)
```

Comment:
Should be made clear it's for the live functionality.
```suggestion
			return e.startLiveQuery(ectx, responseChannel, query, queryContext.TimeRange)
```